### PR TITLE
[operator/dotnet] Add instrumentation docs for dotnet

### PR DESCRIPTION
**Pull request by TylerHelmuth**
Adds a section in the k8s operator instrumentation docs for dotnet.

---

Preview: https://deploy-preview-2565--opentelemetry.netlify.app/docs/k8s-operator/automatic/#net

### CODE REVIEW

**Review by chalin — APPROVED** [review]

File: content/en/docs/k8s-operator/automatic.md [hunk 1]
```diff
@@ -108,7 +108,65 @@ is required for auto-instrumentation to work properly.
...
+  propagators:
+    - tracecontext
+    - baggage
+  sampler:
+    type: parentbased_traceidratio
+    argument: "1"
+EOF
+```
+
+By default, the Instrumentation resource that auto-instruments .Net services
```
Comment (cijothomas) [inline content/en/docs/k8s-operator/automatic.md:132]:
```suggestion
By default, the Instrumentation resource that auto-instruments .NET services
```

File: content/en/docs/k8s-operator/automatic.md [hunk 2]
```diff
@@ -108,7 +108,65 @@ is required for auto-instrumentation to work properly.
...
+spec:
+  exporter:
+    endpoint: http://demo-collector:4317
+  propagators:
+    - tracecontext
+    - baggage
+  sampler:
+    type: parentbased_traceidratio
+    argument: '1'
+  java:
```
Comment (cijothomas) [inline content/en/docs/k8s-operator/automatic.md:160]: java?

File: content/en/docs/k8s-operator/automatic.md [hunk 3]
```diff
@@ -108,7 +108,65 @@ is required for auto-instrumentation to work properly.
 
 ### .NET
 
-Coming Soon
+The following command will create a basic Instrumentation resource that is
+configured specifically for instrumenting .Net services.
```
Comment (cartermp) [inline content/en/docs/k8s-operator/automatic.md:112]:
```suggestion
configured specifically for instrumenting .NET services.
```

File: content/en/docs/k8s-operator/automatic.md (hunk 2 above)
Comment (cartermp) [reply content/en/docs/k8s-operator/automatic.md:160]: Yep, this one has to change 😄

**Review by cartermp — CHANGES_REQUESTED** [review]

File: content/en/docs/k8s-operator/automatic.md [hunk 4]
```diff
@@ -108,7 +108,65 @@ is required for auto-instrumentation to work properly.
...
+```
+
+By default, the Instrumentation resource that auto-instruments .Net services
+uses `otlp` with the `http/proto` protocol. This means that the configured
+endpoint must be able to receive OTLP over `http/proto`. Therefore, the example
+uses `http://demo-collector:4318`, which will connect to the `http` port of the
+otlpreceiver of the Collector created in the previous step.
+
+By default, the .NET auto-instrumentation ships with
+[many instrumentation libraries](/docs/instrumentation/net/libraries/#available-instrumentation-libraries).
```
Comment (pellared) [inline content/en/docs/k8s-operator/automatic.md:139]: This is a lot for "SDK based" instrumentation libraries.
The libraries supported by auto-instrumentation can be found here: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/main/docs/config.md#instrumentations

File: content/en/docs/k8s-operator/automatic.md [hunk 5]
```diff
@@ -108,7 +108,65 @@ is required for auto-instrumentation to work properly.
...
+EOF
+```
+
+By default, the Instrumentation resource that auto-instruments .Net services
+uses `otlp` with the `http/proto` protocol. This means that the configured
+endpoint must be able to receive OTLP over `http/proto`. Therefore, the example
+uses `http://demo-collector:4318`, which will connect to the `http` port of the
+otlpreceiver of the Collector created in the previous step.
+
+By default, the .NET auto-instrumentation ships with
```
Comment (pellared) [inline content/en/docs/k8s-operator/automatic.md:138]: How about adding a hyperlink to https://opentelemetry.io/docs/instrumentation/net/automatic/ ?

File: content/en/docs/k8s-operator/automatic.md [hunk 6]
```diff
@@ -108,7 +108,65 @@ is required for auto-instrumentation to work properly.
...
+  java:
+    env:
+      - name: OTEL_DOTNET_AUTO_TRACES_GRPCNETCLIENT_INSTRUMENTATION_ENABLED
+        value: false
+      - name: OTEL_DOTNET_AUTO_METRICS_PROCESS_INSTRUMENTATION_ENABLED
+        value: false
+```
+
+For more details, see
+[.Net Auto Instrumentation configuration](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/main/docs/config.md#instrumentations).
```
Comment (Kielek) [inline content/en/docs/k8s-operator/automatic.md:169]: .Net -> .NET

File: content/en/docs/k8s-operator/automatic.md [hunk 7]
```diff
@@ -108,7 +108,65 @@ is required for auto-instrumentation to work properly.
...
+the library.
+
+```yaml
+apiVersion: opentelemetry.io/v1alpha1
+kind: Instrumentation
+metadata:
+  name: demo-instrumentation
+spec:
+  exporter:
+    endpoint: http://demo-collector:4317
```
Comment (pavolloffay) [inline content/en/docs/k8s-operator/automatic.md:153]: also the port should change to 4318 for dotnet

File: content/en/docs/k8s-operator/automatic.md (hunk 2 above)
Comment (TylerHelmuth) [reply content/en/docs/k8s-operator/automatic.md]: Fixed

File: content/en/docs/k8s-operator/automatic.md (hunk 6 above)
Comment (TylerHelmuth) [reply content/en/docs/k8s-operator/automatic.md]: Fixed

File: content/en/docs/k8s-operator/automatic.md (hunk 7 above)
Comment (TylerHelmuth) [reply content/en/docs/k8s-operator/automatic.md]: Fixed

File: content/en/docs/k8s-operator/automatic.md (hunk 4 above)
Comment (TylerHelmuth) [reply content/en/docs/k8s-operator/automatic.md]: Changed to link to the repo config doc.

File: content/en/docs/k8s-operator/automatic.md (hunk 5 above)
Comment (TylerHelmuth) [reply content/en/docs/k8s-operator/automatic.md:138]: Updated the final link to link to the site's automatic docs.  This aligns it with the other sections.

**Review by cartermp — APPROVED** [review]

File: content/en/docs/k8s-operator/automatic.md [hunk 8]
```diff
@@ -108,7 +108,65 @@ is required for auto-instrumentation to work properly.
...
+    - baggage
+  sampler:
+    type: parentbased_traceidratio
+    argument: "1"
+EOF
+```
+
+By default, the Instrumentation resource that auto-instruments .NET services
+uses `otlp` with the `http/proto` protocol. This means that the configured
+endpoint must be able to receive OTLP over `http/proto`. Therefore, the example
```
Comment (pellared) [inline content/en/docs/k8s-operator/automatic.md:134]:
```suggestion
endpoint must be able to receive OTLP over `http/protobuf`. Therefore, the example
```

per https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#specify-protocol

**Review by pellared — APPROVED** [review]
Just one nit recommendation

File: content/en/docs/k8s-operator/automatic.md (hunk 8 above)
Comment (TylerHelmuth) [reply content/en/docs/k8s-operator/automatic.md]: Nice catch, fixed.

**Pull request merged**